src/js/module/toolbox/sidebar.js

```diff
@@ -1,58 +1,97 @@
+    const objectsProjs = map(this.manifesto().getCopy('toolboxobjects'), ({ projections }) => projections);
+    const objectsData = chain(objectsIds)
+      .zip(objectsProjs)
+      .map(([ objectId, projections ]) => ({ objectId, projections, currentProjection: 0 }))
+      .value();
 
-  initialize: function initialize() {
-    const objectsIds = underscore.range(this.manifesto().getCopy('toolboxobjects').length);
-    const objectsProjs = underscore.map(this.manifesto().getCopy('toolboxobjects'), function(object) {
-      return object.projections;
+    this.collection = new Collection(objectsData, {
+      model: ToolboxModel,
+      app: this.app,
     });
-    const objectsData = underscore.zip(objectsIds, objectsProjs);
-
-    this.collection = new ToolboxCollection(underscore.map(objectsData, function(objectData) {
-      return {
-        objectId: objectData[0],
-        projections: objectData[1],
-        currentProjection: 0,
-      };
-    }), {app: this.app});
-
-    this.objects = this.collection.map(function(toolboxObjectModel) {
-      return new ToolboxViewObject({
-        model: toolboxObjectModel,
-        app: this.app,
-      });
-    }, this);
-    this.app.on(Const.Event.START_PLANTING, function() {
-      this.$el.show();
-    }, this);
-    this.render( this.objects );
-  },
-
-  render: function render(objects) {
-    const $template = jquery('<div />').append(this.template());
-    const $list = $template.find('.plantingjs-toolboxobject-container');
-
-    objects.forEach(function(object) {
-      $list.append(object.$el);
+    this.render();
+    this.app.on(Event.START_PLANTING, () => {
+      this.renewUserActivity(ACTIVITY_TIMEOUT_VALUE * 2);
     });
-    this.$el.append($template);
   },
 
-  hide: function hide() {
-    this.$el.hide();
+  render() {
+    let objects = this.collection.map((model) => {
+      const {
+        projections: [image],
+        objectId,
+      } = model.attributes;
+      const cid = model.cid;
+
+      return { image, objectId, cid };
+    });
+
+    objects = objects.concat(objects, objects, objects);
+    this.$el.html(template({ objects }));
+    this.makeObjectsDraggable();
   },
 
-  show() {
-    this.$el.show();
+  makeObjectsDraggable() {
+    const config = {
+      containment: '.plantingjs-overlay',
+      helper: 'clone',
+      appendTo: '.plantingjs-overlay',
+      zIndex: 10000,
+    };
+
+    this.$el.find('.js-draggable-object')
+      .draggable(config);
   },
-});
 
-module.exports = ToolboxViewSidebar;
+  onDragStart(el) {
+    const $el = jquery(el.currentTarget);
+    const cid = $el.data('cid');
+    const model = this.collection.get(cid).clone();
+
+    $el.data('model', model.toJSON());
+  },
+
+  onMouseEnter() {
+    this.mouseOver = true;
+    this.renewUserActivity();
+  },
+
+  onMouseLeave() {
+    this.mouseOver = false;
+    this.renewUserActivity();
+  },
+
+  renewUserActivity(timeout = ACTIVITY_TIMEOUT_VALUE) {
```

Comment:
In some cases we want to show sidebar for little longer - fox example on start planting (line 35).